xclim/core/bootstrapping.py

```diff
@@ -102,82 +108,108 @@ def bootstrap_func(compute_indice_func: Callable, **kwargs) -> xarray.DataArray:
+    if xclim.core.utils.uses_dask(da):
+        chunking = {d: "auto" for d in da.dims}
+        chunking["time"] = -1  # no chunking on time to use map_block
+        da = da.chunk(chunking)
+    # overlap of studied `da` and reference period used to compute percentile
+    overlap_da = da.sel(time=slice(*clim))
+    if len(overlap_da.time) == len(da.time):
+        raise KeyError(
+            "`bootstrap` is unnecessary when all years are overlapping between reference "
+            "(percentiles period) and studied (index period) periods"
+        )
+    if len(overlap_da) == 0:
+        raise KeyError(
+            "`bootstrap` is unnecessary when no year overlap between reference "
+            "(percentiles period) and studied (index period) periods."
+        )
     pdoy_args = dict(
-        window=per.attrs["window"],
-        alpha=per.attrs["alpha"],
-        beta=per.attrs["beta"],
-        per=percentile if np.isscalar(percentile) else percentile[0],
+        window=per_da.attrs["window"],
+        alpha=per_da.attrs["alpha"],
+        beta=per_da.attrs["beta"],
+        per=per_da.percentiles.data[()],
     )
-
+    bfreq = _get_bootstrap_freq(kwargs["freq"])
     # Group input array in years, with an offset matching freq
-    freq = kwargs["freq"]
-    _, base, start_anchor, anchor = parse_offset(freq)
+    overlap_years_groups = overlap_da.resample(time=bfreq).groups
+    da_years_groups = da.resample(time=bfreq).groups
+    per_template = per_da.copy(deep=True)
+    acc = []
+    # Compute bootstrapped index on each year of overlapping years
+    for year_key, year_slice in da_years_groups.items():
+        kw = {da_key: da.isel(time=year_slice), **kwargs}
+        if _get_year_label(year_key) in overlap_da.get_index("time").year:
+            # If the group year is in both reference and studied periods, run the bootstrap
+            bda = build_bootstrap_year_da(overlap_da, overlap_years_groups, year_key)
+            if BOOTSTRAP_DIM not in per_template.dims:
+                per_template = per_template.expand_dims(
+                    {BOOTSTRAP_DIM: np.arange(len(bda._bootstrap))}
+                )
+                if xclim.core.utils.uses_dask(bda):
+                    chunking = {
+                        d: bda.chunks[bda.get_axis_num(d)]
+                        for d in set(bda.dims).intersection(set(per_template.dims))
+                    }
+                    per_template = per_template.chunk(chunking)
+            per = xr.map_blocks(
+                percentile_doy.__wrapped__,  # strip history update from percentile_doy
+                obj=bda,
+                kwargs={**pdoy_args, "copy": False},
+                template=per_template,
+            )
+            if "percentiles" not in per_da.dims:
+                per = per.squeeze("percentiles")
+            kw[per_key] = per
+            value = compute_index_func(**kw).mean(dim="_bootstrap", keep_attrs=True)
```

Comment:
```suggestion
            value = compute_index_func(**kw).mean(dim=BOOTSTRAP_DIM, keep_attrs=True)
```
A small oubli I devine.

Reply:
You devine well, merci.